diff --git a/mythmusic/mythmusic/musicplayer.cpp b/mythmusic/mythmusic/musicplayer.cpp
index b0cccf5..d94e08f 100644
--- a/mythmusic/mythmusic/musicplayer.cpp
+++ b/mythmusic/mythmusic/musicplayer.cpp
@@ -105,6 +105,7 @@ MusicPlayer::~MusicPlayer()
         m_cdWatcher->stop();
         m_cdWatcher->wait();
         delete m_cdWatcher;
+        m_cdWatcher = NULL;
     }
 
     if (!hasClient())
@@ -283,6 +284,12 @@ void MusicPlayer::stop(bool stopAll)
         m_output = NULL;
     }
 
+    if (stopAll && m_cdWatcher)
+    {
+        m_cdWatcher->stop();
+        m_cdWatcher->wait();
+    }
+
     // because we don't actually stop the audio output we have to fake a Stopped
     // event so any listeners can act on it
     OutputEvent oe(OutputEvent::Stopped);
diff --git a/mythmusic/mythmusic/polygon.h b/mythmusic/mythmusic/polygon.h
index 2884d8e..d859d5d 100644
--- a/mythmusic/mythmusic/polygon.h
+++ b/mythmusic/mythmusic/polygon.h
@@ -18,13 +18,13 @@ class Bitmap
         delete[] data;
         width = w;
         height = h;
-        data = new Pixel[w*h+extra];
+        data = new Pixel[2*w*h+extra];
         clear();
     }
   
     void clear()
     {
-        memset(data,0,sizeof(Pixel)*(width*height+extra));
+        memset(data,0,sizeof (Pixel)*(2*width*height+extra));
     }
 };
 #endif
